client/src/planwise/client/scenarios/handlers.cljs

```diff
@@ -32,6 +32,40 @@
  (fn [db [_ path value]]
    (assoc-in db path value)))
 
+;; Views stack
+
+(rf/reg-event-db
+ :scenarios/push-view
+ in-scenarios
+ (fn [db [_ view]]
+   (println [".......x" view (:views-stack db)])
+   (update-in db [:views-stack] #(conj % view))))
+
+(rf/reg-event-db
+ :scenarios/pop-view
+ in-scenarios
+ (fn [db [_]]
+   (println [".......z" (:views-stack db)])
+   (update-in db [:views-stack] #(if (empty? %) % (pop %)))))
+
+(rf/reg-event-db
+ :scenarios/reset-state
+ in-scenarios
+ (fn [db [_]]
+   (assoc-in db [:views-stack] [])))
+
```

Comment:
Manipulating the stack does not need to be done in separate events. You can extract the push/pop/reset functions, make them take the `db` as a first parameter and thread them with the rest of the `db` updates in the existing events. Eg.
```
(defn push-view
  [db new]
  (update-in db [:views-stack] #(conj % view)))

(defn pop-view
  [db]
  (update-in db [:views-stack] #(if (empty? %) % (pop %))))

(defn reset-view-state
  [db]
  (assoc-in db [:views-stack] []))
```

Then use it, for example:
```
(rf/reg-event-fx
  :scenarios/cancel-dialog
  (fn [{:keys [db]} [_]]
   {:db (-> db
            (assoc :changeset-dialog nil
                   :rename-dialog nil)
            (pop-view))}))
```
In short, make your own view stack manipulation functions and use them to mutate the db.

Reply:
I'll do it.